core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -352,13 +348,15 @@ class LogManager(logDirs: Seq[File],
         val numLogsLoaded = new AtomicInteger(0)
         numTotalLogs += logsToLoad.length
 
-        val jobsForDir = logsToLoad.map { logDir =>
+        val jobsForDir = logsToLoad
+          .filter(logDir => Log.parseTopicPartitionName(logDir).topic != KafkaRaftServer.MetadataTopic)
```

Comment:
Will the metadata topic get passed into LogManager? I would guess not since it's a Raft topic and not a regular Kafka topic. 

Also style nit: you can do `logsToLoad.filter { logDir => ... }` similar to the `map` below

Reply:
Yeah, we use the list of topics just to know what to check for log config overrides.  This will see anything in the log dir.  Now that I realize it, we could probably use this to know what topics we need to look at for log config overrides.  But I don't want to change too much at this point.

Reply:
We could get rid of the parameter in `startup()` and create a 'TopicLogConfigOverrideRetriever` class that held the `ConfigRepository` and would retrieve any topic configs in a thread-safe manner -- then we just pass that to `loadLog()` and it asks for the overrides, and the class would retrieve iff they haven't been retrieved before.  WDYT?

Reply:
I suspect we shouldn’t load topic configs on-demand because it would change the current behavior.  Specifically, if we can’t load the configs, currently the object instantiation will raise an exception.  The current patch will cause the exception to be raised in startup(), which is effectively the same thing.  But if we load them on-demand then both instantiation and startup() will succeed, and the exception will occur in a thread doing recovery.